bitcoin/src/lib.rs

```diff
@@ -880,6 +893,46 @@ impl BitcoinCoreApi for BitcoinCore {
         Ok(())
     }
 
+    async fn rescan_electrs_for_addresses<A: PartialAddress + Send + Sync + 'static>(
+        &self,
+        addresses: Vec<A>,
+    ) -> Result<(), Error> {
+        for address in addresses.into_iter() {
+            let address = address.encode_str(self.network)?;
+            let all_transactions =
+                esplora_btc_api::apis::address_api::get_address_tx_history(&self.electrs_config, &address).await?;
```

Comment:
Can you add a TODO note that this should be paged? I think this call will fail if there are too many addresses.

Reply:
Actually, you're right (assuming you meant too many transactions). I operated under the assumption that each issue address will only ever be used once, but that does open up an attack where anyone can spam 25 dust-amount transactions to an issue request, and then this API call will no longer pick up the original payment.

Only an issue if the vault client isn't running in the first place while the real issue transaction occurs, but definitely should be mitigated, I'll work on adding payment. Good catch, thanks.